### ISSUE TITLE
Make the loading icon accessibility compliant

### ISSUE THREAD

**Issue by Alex-Bubblemaster** [telerik/kendo-ui-core#4874]
### Enhancement

Ticket ID: 1398151 

- [x] DropDownTree
- [x] Grid
- [x] ListBox
- [x] ListView
- [x] PivotGrid
- [x] Scheduler
- [x] Spreadsheet
- [x] TreeList
- [x] TreeView

### Reproduction of the problem

Screen readers should be notified that the data is loading for databound widgets.

### Current behavior

The MultiSelect, ComboBox, AutoComplete and List already implement this feature with the help of the `aria-busy="true"` attribute. The rest of the widgets should also notify screen reader users that they are loading.

### Expected/desired behavior

Instead of providing this functionality for all widgets separately, extend the kendo.ui.progress method to decorate the div with the attribute and then use the method in the source of the widgets if possible.

### Possible Workarounds

1. Extend the kendo.ui.progress method and declare it before any Kendo Ui Widget is initialized

[Dojo](https://dojo.telerik.com/EtEFUxad/2)

```
$.extend(kendo.ui, {
        progress: function(container, toggle, options) {
            var mask = container.find(".k-loading-mask"),
                support = kendo.support,
                browser = support.browser,
                isRtl, leftRight, webkitCorrection, containerScrollLeft, cssClass;

                options = $.extend({}, {
                    width: "100%",
                    height: "100%",
                    top: container.scrollTop(),
                    opacity: false
                }, options);

                cssClass = options.opacity ? 'k-loading-mask k-opaque' : 'k-loading-mask';

            if (toggle) {
                if (!mask.length) {
                    isRtl = support.isRtl(container);
                    leftRight = isRtl ? "right" : "left";
                    containerScrollLeft = container.scrollLeft();
                    webkitCorrection = browser.webkit ? (!isRtl ? 0 : container[0].scrollWidth - container.width() - 2 * containerScrollLeft) : 0;

                    mask = $(kendo.format("<div class='{0}'><span class='k-loading-text' role='alert' aria-busy='alert'>{1}</span><div class='k-loading-image'/><div class='k-loading-color'/></div>", cssClass, "Please wait data is loading"))
                        .width(options.width).height(options.height)
                        .css("top", options.top)
                        .css(leftRight, Math.abs(containerScrollLeft) + webkitCorrection)
                        .prependTo(container);
                }
            } else if (mask) {
                mask.remove();
            }        
        }
      });
```
2) Use a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)

[Dojo](https://dojo.telerik.com/EtEFUxad)

```
const target = document.body;

      // config object
      const mutationConfig = {
        childList: true,
        subtree: true
      };

      // subscriber function
      function subscriber(mutations) {
        mutations.forEach((mutation) => {
          // handle mutations here
          if($(mutation.addedNodes[0]).hasClass("k-loading-mask")){
            $("span.k-loading-text").attr("role", "alert");
            $("span.k-loading-text").attr("aria-busy", "alert");
            $("span.k-loading-text").text("Please wait data is loading");
          }
        });
      }

      // instantiating observer
      const observer = new MutationObserver(subscriber);

      // observing target
      observer.observe(target, mutationConfig);
```

### Environment

* **Kendo UI version:** 2019.1.220

**Comment by GaloisGirl** [comment]
The first solution gives me the error  "Cannot read property 'loading' of undefined". That's because `kendo.ui.progress` is used as a dictionary in https://github.com/telerik/kendo-ui-core/blob/master/src/kendo.core.js#L3310 and as a function in https://github.com/telerik/kendo-ui-core/blob/master/src/kendo.core.js#L3211 .

I work around it by adding `$.extend(kendo.ui.progress, { messages: {loading: ''}})` , but the way this code is written is bad practice.
